canonical-web-and-design/maas-ui
# fix(ui): correctly filter list of power types for use in Add Chassis form

## Done

- Correctly filter the add chassis form power types to only use those that have `can_probe === true`, which is the correct identifier for power types that can use the `add_chassis` http method.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open the `Add Chassis` form from the machine list
- Check that the list of power types are only the ones that have `can_probe`. It should be similar to the [old hardcoded list](https://github.com/canonical-web-and-design/maas-ui/blob/dc9ec5c7dff910ceec8b9bd99f58a561775b0858/ui/src/app/machines/views/AddChassis/chassisPowerTypes.js).

## Fixes

Fixes #1285

## Launchpad issue

[lp#1876855](https://bugs.launchpad.net/maas/+bug/1876855)
